src/_posts/databases/mysql/2000-01-01-troubleshooting.md

```diff
@@ -0,0 +1,169 @@
+consists in identifying the queries that are the most time-consuming.
+
+The default MySQL® configuration provided by Scalingo logs slow queries. A
+query is considered slow if it takes more than 2 seconds to execute. So, the
+very first action to take when chasing expensive queries would be to find
+problematic query statements by [viewing your database logs]({% post_url databases/mysql/2000-01-01-monitoring %}#inspecting-database-logs).
+
+Here is an example of a slow query log:
+
+```text
+2024-03-07 16:02:11.224378809 +0100 CET [mysql-1] # Time: 2024-03-07T15:02:11.223955Z
+2024-03-07 16:02:11.224396794 +0100 CET [mysql-1] # User@Host: my_app_4553[my_app_4553] @  [192.168.100.3]  Id: 23004
+2024-03-07 16:02:11.224397505 +0100 CET [mysql-1] # Query_time: 10.000321  Lock_time: 0.000000 Rows_sent: 1  Rows_examined: 1
+2024-03-07 16:02:11.224398023 +0100 CET [mysql-1] SET timestamp=1709823725;
+2024-03-07 16:02:11.224398826 +0100 CET [mysql-1] SELECT SLEEP(10);
+```
+
+### Identifying a Low Cache Hit Ratio
+
+[As seen previously](#understanding-memory-consumption), relying on swap usage
+and excessive disk reads can significantly impact your database performance.
+Consequently, measuring the number of times the database has to read data from
+the disk instead of fetching it from the cache (stored in memory) can help
+identify non-optimal queries or inappropriate configuration.
+
+The following instructions allow to compute the *cache hit ratio*: a
+measurement that quantifies the proportion of queries fully satisfied by only
+hitting the cache.
+
+A high cache hit ratio indicates that a significant portion of the required
+data is already in the cache, leading to efficient query performance. On the
+other hand, a low cache hit ratio suggests that the query is not sufficiently
+benefiting from cached data, potentially resulting in slower performance and
+more disk I/O operations.
+
+**For queries that run very oftently**, the cache hit ratio should ideally
+neighbor `100%`.
+
+1. Make sure you have correctly [setup the Scalingo command line tool]({% post_url platform/cli/2000-01-01-start %})
+2. From the command line, open a console for your MySQL® addon: 
+   ```bash
+   scalingo --app my-app mysql-console
+   ```
+3. Run the following query:
+   ```sql
+   SHOW GLOBAL STATUS;
+   ```
+4. Locate the values for `Innodb_buffer_pool_read_requests` and
+   `Innodb_buffer_pool_reads`
+   - `Innodb_buffer_pool_read_requests` indicates the number of logical reads
+     InnoDB has done ;
+   - `Innodb_buffer_pool_reads` indicates the number of logical reads InnoDB
+     could not satisfy from the buffer pool, and had to be done directly from
+     the disk (physical reads).
+5. Compute the percentage of queries that are done from the cache pool:
+   ```text
+   Cache Hit Ratio = (Innodb_buffer_pool_read_requests - Innodb_buffer_pool_reads)
+                     * 100.0 / Innodb_buffer_pool_read_requests
+   ```
```

Comment:
question: I'm pretty sure it could be possible to compute this using the CLI, don't you think? It could be nice to provide a one-liner to compute this

Reply:
Indeed! The one-liner looks like this:
```sql
SELECT ((g1.VARIABLE_VALUE - g2.VARIABLE_VALUE) * 100.0 / g1.VARIABLE_VALUE) AS ratio
FROM performance_schema.global_status g1
INNER JOIN performance_schema.global_status g2
WHERE g1.VARIABLE_NAME = 'Innodb_buffer_pool_read_requests'
and g2.VARIABLE_NAME = 'Innodb_buffer_pool_reads';
```
What do you think about it?

Reply:
This works fine!! You can add it :wink:

Reply:
Done in https://github.com/Scalingo/documentation/pull/2501/commits/09fc4ab3196579cb95c84c57571051653df483fe